commands/peers/peer-rpc-clnt.go

```diff
@@ -0,0 +1,120 @@
+package peercommands
+
+import (
+	"fmt"
+
+	"github.com/gluster/glusterd2/peer"
+
+	log "github.com/Sirupsen/logrus"
+	config "github.com/spf13/viper"
+	netctx "golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+var (
+	opRet   int32
+	opError string
+)
+
+// ValidateAddPeer is the validation function for AddPeer to invoke the rpc
+// server call
+func ValidateAddPeer(p *peer.PeerAddRequest) (*PeerAddResp, error) {
+	args := &PeerAddReq{Name: p.Name, Addresses: p.Addresses}
+
+	rsp := new(PeerAddResp)
+	remoteAddress := fmt.Sprintf("%s:%s", p.Name, config.GetString("rpcport"))
+	rpcConn, e := grpc.Dial(remoteAddress, grpc.WithInsecure())
+	if e != nil {
+		log.WithField("error", e).Error("net.Dial() call failed")
+		opRet = -1
+		opError = e.Error()
+		rsp.OpRet = opRet
```

Comment:
Yup. I need to fix this. This file was moved from the previous location in `rpc/client` with just changes required to compile.